sdk/core/azure-core-http-vertx/src/main/java/com/azure/core/http/vertx/implementation/VertxRequestWriteSubscriber.java

```diff
@@ -101,6 +102,10 @@ private void write(ByteBuffer bytes) {
                 }
             } else {
                 this.state = State.ERROR;
+                if (error != null) {
```

Comment:
it feels like this check should be a 1-line static helper function we use everywhere

Reply:
Do you mean across all our Subscriber implementations or this one specifically?